src/Locator/FileNotFound.php

```diff
@@ -33,25 +33,37 @@
 
 declare(strict_types=1);
 
-namespace Infection\Finder\Exception;
+namespace Infection\Locator;
+
+use function implode;
+use RuntimeException;
+use function Safe\sprintf;
+use Webmozart\Assert\Assert;
 
 /**
  * @internal
  */
-final class LocatorException extends \RuntimeException
+final class FileNotFound extends RuntimeException
 {
-    public static function fileOrDirectoryDoesNotExist(string $name): self
+    /**
+     * @param string[] $roots
+     */
+    public static function createForFile(string $file, array $roots): self
```

Comment:
Perhaps `fromFilename()` is a better choice, along with `$fileName` as parameter name?

Reply:
I like it

Reply:
except it's `$file` or maybe `$filePath`  since in this peculiar case it's the path of the file not its name